src/primitives/transaction.h

```diff
@@ -12,6 +12,12 @@
 #include <serialize.h>
 #include <uint256.h>
 
+/**
+ * A flag that is ORed into the protocol version to designate that a transaction
+ * should be (un)serialized without witness data.
+ * Make sure that this does not collide with any of the values in `version.h`
+ * or with `ADDRv2_FORMAT`.
+ */
```

Comment:
Potential followup: perhaps there is something cleaner we can do in the future for serialization/deserialization options other than abusing the protocol version.

Reply:
Right, there is room for improvement here. I opened https://github.com/bitcoin/bitcoin/issues/19477 to track this.